src/guest_memory.rs

```diff
@@ -407,6 +413,9 @@ pub trait GuestMemory {
     /// Type of objects hosted by the address space.
     type R: GuestMemoryRegion;
 
+    /// Lifetime generic associated iterator.
+    type Iterator: for<'a> GuestMemoryIterator<'a, Self::R>;
```

Comment:
Can we rename the associated type to `I` so it's more consistent with the already existing `R`? Calling it `Iterator` is also a bit confusing, since it's not the actual type that's used as the iterator.